modules/map/relative_map/navigation_lane.cc

```diff
@@ -162,20 +166,24 @@ void NavigationLane::MergeNavigationLineAndLaneMarker(common::Path *path,
   const double len = std::fmin(
       navigation_path.path_point(navigation_path.path_point_size() - 1).s(),
       lane_marker_path.path_point(lane_marker_path.path_point_size() - 1).s());
+  const double start_s = std::fmax(navigation_path.path_point(0).s(),
+                                   lane_marker_path.path_point(0).s());
 
   const double ds = 1.0;
   int navigation_index = 0;
   int lane_marker_index = 0;
-  for (double s = 0.0; s < len; s += ds) {
+  common::Path temp_path;
+  for (double s = start_s; s < len; s += ds) {
     auto p1 = GetPathPointByS(navigation_path, navigation_index, s,
                               &navigation_index);
     auto p2 = GetPathPointByS(lane_marker_path, lane_marker_index, s,
                               &lane_marker_index);
-    auto *p = path->add_path_point();
+    auto *p = temp_path.add_path_point();
```

Comment:
can we do the modification directly on path?

Reply:
It may not be appropriate. Because directly modifying the points of the original navigation path will cause the starting and ending points of the original path may not be modified, and ultimately the resulting navigation path will not be smooth.  Also, the length of a navigation path does not exceed 250 meters, so using temporary variables does not reduce much efficiency.
Here is the implementation code based on your idea. The result is as follows. It can be seen that the lane boundary is not particularly smooth.

``` cpp
  const double ds = 1.0;
  int navigation_index = 0;
  int lane_marker_index = 0; 
  for (double s = start_s; s < len; s += ds) {
    auto p1 = GetPathPointByS(navigation_path, navigation_index, s,
                              &navigation_index);
    auto p2 = GetPathPointByS(lane_marker_path, lane_marker_index, s,
                              &lane_marker_index);
    // auto *p = temp_path.add_path_point();
    common::PathPoint *p = nullptr;
    if (path->path_point_size() > 0) {
      p = path->mutable_path_point(navigation_index);
    } else {
      p = path->add_path_point();
    }
    const double kWeight = 0.9;
    *p = common::util::GetWeightedAverageOfTwoPathPoints(p1, p2, kWeight,
                                                         1 - kWeight);
  }  
```
![screenshot from 2018-06-04 15-58-58](https://user-images.githubusercontent.com/31435476/40905867-c9eb26a2-6811-11e8-9be0-84c90e2e70a3.png)